File: src/main/java/com/basil/shoppingcart/service/OrderService.java
package com.basil.shoppingcart.service;

import java.util.List;

import com.basil.shoppingcart.dto.response.OrderResponse;

public interface OrderService {

    OrderResponse checkout();

    List<OrderResponse> getMyOrders();

    OrderResponse getOrder(Long orderId);

    List<OrderResponse> getAllOrders();
    OrderResponse updateOrderStatus(Long orderId,String status);
}

File: shopping-cart-frontend/src/services/orderService.js
import apiClient from "./apiClient";

const orderService = {

    createOrder: async (

        orderData

    ) => {

        const response =

            await apiClient.post(

                "/orders/checkout",

                orderData

            );


        return response.data;

    },


    getMyOrders: async () => {

        const response =

            await apiClient.get(

                "/orders"

            );


        return response.data;

    },


    getOrderById: async (

        orderId

    ) => {

        const response =

            await apiClient.get(

                `/orders/${orderId}`

            );


        return response.data;

    }

};


export default orderService;